lesson7-questions.txt

```diff
@@ -87,19 +111,37 @@ You will see the SQL that is executed logged to the screen.  What is that SQL?
 There is a security exposure, a risk that a hacker could do damage.  What is that
 risk? What two things are done in your code to prevent the risk?
 
+The risk is that someone could delete a post that was created by someone else. That should not
+be allowed. The code should check if the person is logged on and if it is their post before they
+are allowed to delete it.
+I think this is the relevant code:
+before_action :set_post, only: %w[show edit update destroy]
+  before_action :check_access, only: %w[edit update delete]
+Any idea why one says destroy and one says delete, even though I have no method called delete?
+It also sets the post on destroy to make sure you are deleting the right one, correct?
 
 
 9. The routes for subscriptions are nested under the forum routes.  How does this
 change the routes?  What is the additional hash key and value that is passed in the
 params hash to the subscriptions controller as a result of this nesting? Which routes
 are unaffected by the nesting?
 
+When you nest subscriptions under forums, the routes become hierarchical, meaning the routes for
+subcriptions will include the forum context in their URL structure. In this case, the new and create route
+will include the forum_id, but the show, edit, update,and destroy will not because of the shallow: true?
+The index route is unaffected by the nesting because of except: [:index]? This is the relevant code, I believe:
```

Comment:
Correct, the `except` excludes the actions in the array from being created under the nesting.